src/main/java/org/traccar/api/resource/SessionResource.java

```diff
@@ -99,9 +99,15 @@ public User get(@QueryParam("token") String token) throws SQLException, Unsuppor
     @PermitAll
     @POST
     public User add(
-            @FormParam("email") String email, @FormParam("password") String password) throws SQLException {
+            @FormParam("email") String email, @FormParam("password") String password,
+            @FormParam("code") int code) throws SQLException {
         User user = Context.getPermissionsManager().login(email, password);
         if (user != null) {
+            if (Context.getConfig().getBoolean("totp.enabled")) {
```

Comment:
Looks like it's checked twice. Here and in the `isTotpAuthCodeValid`.

Reply:
This check has been removed is it is indeed also checked within the user class method isTotpAuthCodeValid.